src/fheroes2/gui/interface_events.cpp

```diff
@@ -132,7 +133,7 @@ void Interface::AdventureMap::EventSwitchFocusedHero( const int32_t tileIndex )
 {
     Heroes * selectedHero = world.GetTiles( tileIndex ).getHero();
     const Heroes * currentHero = GetFocusHeroes();
-    if ( selectedHero == nullptr || selectedHero == currentHero || selectedHero->GetColor() != currentHero->GetColor() ) {
+    if ( selectedHero == nullptr || selectedHero == currentHero || selectedHero->GetColor() != Settings::Get().GetPlayers().getCurrentColor() ) {
```

Comment:
Hi, @zenseii, in the case when `selectedHero == nullptr` we do not ever use the `currentHero`, so we could slightly optimize this part by calling `GetFocusHeroes()` only after the first nullptr check:
```suggestion
    if ( selectedHero == nullptr || selectedHero == GetFocusHeroes()|| selectedHero->GetColor() != Settings::Get().GetPlayers().getCurrentColor() ) {
```
And removing the 135 line.